lib/reindex/s3_bucketd.py

```diff
@@ -216,6 +248,12 @@ def _sum_objects(self, bucket, listing):
                     # bucketd v7 returns an encoded string
                     data = json.loads(obj['value'])
                     size = data.get('content-length', 0)
+
+                if self._only_latest_when_locked and bucket.object_locked and '\x00' in obj['key']:
```

Comment:
I have the feeling checking for a zero-byte is too low-level. We should instead either rely on a `DelimiterVersions` listing for all cases, or change the listing type to `DelimiterMaster` in this particular case. I don't know how easy or hard it would be to change this.

For example in V1 metadata, latest delete markers would only have a versioned key and no master key.

Reply:
That was my original feeling as well, but the feature is only requested for v0. I'm hesitant to do too much work on this as utapi is essentially "finished" and only receiving bug fixes (and small customer driven features like this one). Using a different listing type could be possible, but I can't remember the differences off the top of my head.

Reply:
I get your point, but I don't think we can "trust" customers to never ever use utapi v1 with v1 buckets. Don't forget that v1 buckets are in the corner and already available in 9.2.0.4.

This being said I have no reason to block the PR if we say that utapi v1 is not compatible with metadata v1, but we need to be careful here and we should probably ask @rahulreddy what he thinks from product perspective.